src/container/container.ts

```diff
@@ -108,11 +108,29 @@ class Container implements interfaces.Container {
             };
         };
 
+        let getRebindFunction = (moduleId: string) => {
+            return (serviceIdentifier: interfaces.ServiceIdentifier<any>) => {
+                let _rebind = this.rebind.bind(this);
```

Comment:
Maybe change these two lines to

```
const bindingToSyntax = this.rebind.call(this, serviceIdentifier);
```

As having three word bind three times in a row is a bit of a mind bender! Haha